lib/aws/request.js

```diff
@@ -0,0 +1,224 @@
+  sdk.config.logger = log;
+}
+
+// Use HTTPS Proxy (Optional)
+const proxy =
+  process.env.proxy ||
+  process.env.HTTP_PROXY ||
+  process.env.http_proxy ||
+  process.env.HTTPS_PROXY ||
+  process.env.https_proxy;
+
+const proxyOptions = {};
+if (proxy) {
+  Object.assign(proxyOptions, new URL(proxy));
+}
+
+const ca = process.env.ca || process.env.HTTPS_CA || process.env.https_ca;
+
+let caCerts = [];
+
+if (ca) {
+  // Can be a single certificate or multiple, comma separated.
+  const caArr = ca.split(',');
+  // Replace the newline -- https://stackoverflow.com/questions/30400341
+  caCerts = caCerts.concat(caArr.map((cert) => cert.replace(/\\n/g, '\n')));
+}
+
+const cafile = process.env.cafile || process.env.HTTPS_CAFILE || process.env.https_cafile;
+
+if (cafile) {
+  // Can be a single certificate file path or multiple paths, comma separated.
+  const caPathArr = cafile.split(',');
+  caCerts = caCerts.concat(caPathArr.map((cafilePath) => fs.readFileSync(cafilePath.trim())));
+}
+
+if (caCerts.length > 0) {
+  Object.assign(proxyOptions, {
+    rejectUnauthorized: true,
+    ca: caCerts,
+  });
+}
+
+// Passes also certifications
+if (proxy) {
+  sdk.config.httpOptions.agent = new HttpsProxyAgent(proxyOptions);
+} else if (proxyOptions.ca) {
+  // Update the agent -- http://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-registering-certs.html
+  sdk.config.httpOptions.agent = new https.Agent(proxyOptions);
+}
+
+// Configure the AWS Client timeout (Optional).  The default is 120000 (2 minutes)
+const timeout = process.env.AWS_CLIENT_TIMEOUT || process.env.aws_client_timeout;
+if (timeout) {
+  sdk.config.httpOptions.timeout = parseInt(timeout, 10);
+}
+PromiseQueue.configure(Promise);
+const requestQueue = new PromiseQueue(2, Infinity);
+
+const MAX_RETRIES = (() => {
+  const userValue = Number(process.env.SLS_AWS_REQUEST_MAX_RETRIES);
+  return userValue >= 0 ? userValue : 4;
+})();
+
+const accelerationCompatibleS3Methods = new Set(['upload', 'putObject']);
+
+const shouldS3Accelerate = (method, params) => {
+  if (
+    accelerationCompatibleS3Methods.has(method) &&
+    params &&
+    params.isS3TransferAccelerationEnabled
+  ) {
+    log('Using S3 Transfer Acceleration Endpoint...');
+    return true;
+  }
+  return false;
+};
+
+const getServiceInstance = memoize(
+  (service, method) => {
+    const Service = _.get(sdk, service.name);
+    // we translate params to an object for the service creation by selecting keys of interest
+    const serviceParams = {};
+    if (service.params) {
+      if (service.params.credentials) serviceParams.credentials = service.params.credentials;
+      if (service.params.region) serviceParams.region = service.params.region;
+    }
+    if (service.name === 'S3') {
+      serviceParams.useAccelerateEndpoint = shouldS3Accelerate(method, service.params);
+    }
+    if (_.isEmpty(serviceParams)) {
+      return new Service();
+    }
+    return new Service(serviceParams);
```

Comment:
I gues both calls are equivalent, so it can be just `return new Service(params)`